Extensions/Physics2Behavior/tests/utils.spec.js

```diff
@@ -0,0 +1,104 @@
+// @ts-check
+
+describe('computeCurrentContactsFromStartedAndEndedContacts', () => {
+  it('returns same current contacts if nothing happened', () => {
+    const contacts = ['A', 'B', 'C'];
+    const startedContacts = [];
+    const endedContacts = [];
+    const expectedResolvedContacts = [...contacts];
+    gdjs.physics2.computeCurrentContactsFromStartedAndEndedContacts(
+      contacts,
+      startedContacts,
+      endedContacts
+    );
+    expect(contacts).to.eql(expectedResolvedContacts);
+  });
+
+  it('returns current contacts with started contacts added', () => {
+    const contacts = ['A', 'B', 'C'];
+    const startedContacts = ['Z', 'Q'];
+    const endedContacts = [];
+    const expectedResolvedContacts = [...contacts, ...startedContacts];
+    gdjs.physics2.computeCurrentContactsFromStartedAndEndedContacts(
+      contacts,
+      startedContacts,
+      endedContacts
+    );
+    expect(contacts).to.eql(expectedResolvedContacts);
+  });
+
+  it('returns current contacts with ended contacts removed', () => {
+    const contacts = ['A', 'B', 'C', 'Z'];
+    const startedContacts = [];
+    const endedContacts = ['Z', 'B', 'R'];
```

Comment:
The 'R' can't be removed if it doesn't exist. Every case that can't happens should probably be in a specific describe to make it easier for the reader to understand how the behavior works.

Reply:
I'm not keen on listing things that can't happen somewhere because some things will be forgotten and it can be misleading in the future.